src/pretty/plain.rs

```diff
@@ -0,0 +1,110 @@
+use pretty::DocAllocator;
+use typst_syntax::{ast::AstNode, SyntaxKind, SyntaxNode};
+
+use super::{doc_ext::DocExt, flow::FlowStylist, ArenaDoc, PrettyPrinter};
+
+#[derive(Debug)]
+enum PlainItem<'a> {
+    Item(ArenaDoc<'a>),
+    Comma,
+    Linebreak(usize),
+    LineComment(ArenaDoc<'a>),
+    BlockComment(ArenaDoc<'a>),
+}
+
+/// A stylist that keeps the original structure.
+pub struct PlainStylist<'a> {
```

Comment:
Just wondering: we are having multiple stylist now. Do they have similar interface? If so, we can add a trait for Stylist

Reply:
I don't have an idea of how to better abstract and reuse these stylists at the moment.
The `FlowStylist` is the simplest one that can only handle spacings, and can be used elsewhere.
The `PlainStylist` is like `ListStylist` (accepts an iterable and creates a doc), but doesn't provide flexible fitting. I think it's not time for further refactoring yet.